src/UpdateWizard.js

```diff
@@ -0,0 +1,285 @@
+// (c) Copyright 2018 SUSE LLC
+/**
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+**/
+import React from 'react';
+import './styles/deployer.less';
+import { STATUS } from './utils/constants.js';
+import WizardProgress from './components/WizardProgress';
+import { Map } from 'immutable';
+import InstallWizard from  './InstallWizard.js';
+import {
+  COMMON_GLOBAL_STATES_VARS, COMMON_PERSISTED_STATES_VARS,
+  UPDATE_GLOBAL_STATES_VARS, UPDATE_PERSISTED_STATES_VARS
+} from "./utils/constants";
+
+/**
+ * The UpdateWizard component is a container for loading update pages which will launch
+ * processes pages. It also managing ordering the update process pages and tracking
+ * process state across them.
+ * TODO
+ * this is the first step to move update related processes out, more refactoring
+ * will come
+ */
+class UpdateWizard extends InstallWizard {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      // The current step in the wizard
+      currentStep: undefined,
+
+      // The status and name of each step in the wizard.
+      // when update, the steps are unknown to start with until user
+      // startUpdate process. steps are used to record step name and step progress
+      // steps are persisted
+      steps: undefined,
+
+      // the current page where user is at hopefully, in the future it can be
+      // used to return to the page where user gets session timeout
+      currentMenuName: this.props.menuName,
+
+      // the menu name that has process in progress, this can be different from
+      // currentMenuName
+      processMenuName: undefined,
+
+      // the operation name that is process in progress
+      processOperation: undefined,
+
+      // when update, the pages are dynamically assembled based on steps when
+      // user startUpdate process. pages include names and page components
+      // which are needed when render html pages
+      // pages are not persisted
+      pages: undefined,
+
+      // operationProps record all the bits that need to run to update
+      // TODO how to deal with install password
+      operationProps: undefined,
+    };
+
+    // Indicate which of the above state variables are passed to wizard pages and can be set by them
+    this.globalStateVars =
```

Comment:
Instead of having all of these global constants (which expose inner details of these classes too broadly), it would be better to do:
`this.globalStateVars = this.globalStateVars.concat(['currentMenuName, 'processMenuName' ...]);`

Reply:
k, will look into it.